modules/imap/site.js

```diff
@@ -1375,3 +1380,23 @@ if (message) {
         childList: true
     });
 }
+
+const handleDownloadMsgSource = function() {
+    const messageSource = document.querySelector('pre.msg_source');
+    const blob = new Blob([messageSource.textContent], { type: "message/rfc822" });
+    const url = URL.createObjectURL(blob);
+    const a = document.createElement('a');
+    a.href = url;
+    const subject = messageSource.textContent.match(/Subject: (.*)/)?.[1];
+    a.download = subject + '.eml';
```

Comment:
Can we fallback to msg uid or something similar if this regex match fails, so the filename is still something valid?

Reply:
That's a valid point because the subject could be empty. So, let's use the msg uid as a fallback.